src/js/states/mods.js

```diff
@@ -132,6 +143,28 @@ export class ModsState extends TextualGameState {
         this.app.platformWrapper.openExternalLink(THIRDPARTY_URLS.modBrowser);
     }
 
+    toggleMod(checkbox, modID) {
+        if (checkbox.classList.contains("checked")) {
+            MODS.disableMod(modID);
+        } else {
+            MODS.enableMod(modID);
+        }
+        checkbox.classList.toggle("checked");
+    }
+
+    async applyModToggle() {
+        const signals = this.dialogs.showWarning(
+            T.dialogs.applyModsWarning.title,
+            T.dialogs.applyModsWarning.desc,
+            ["cancel:good", "continue:bad"]
+        );
+        await new Promise(resolve => {
+            signals.continue.add(_ => {
+                MODS.applyModEnableDisable();
+            });
+        });
```

Comment:
this never invokes `resolve` (and although promisifying things is nice, it might be better to not until it's actually needed)